core/arch/riscv/kernel/sbi.c

```diff
@@ -29,6 +29,23 @@ struct sbiret {
 
 #define sbi_ecall(...) _sbi_ecall(__VA_ARGS__, 0, 0, 0, 0, 0, 0, 0)
 
+/**
+ * sbi_probe_extension() - Check if an SBI extension ID is supported or not.
+ * @extid: The extension ID to be probed.
+ *
+ * Return: 1 or an extension specific nonzero value if yes, 0 otherwise.
+ */
+int sbi_probe_extension(int extid)
+{
+	struct sbiret ret;
+
+	ret = sbi_ecall(SBI_EXT_BASE, SBI_EXT_BASE_PROBE_EXT, extid);
+	if (!ret.error)
+		return ret.value;
```

Comment:
for consistency with the inline description, I suggest to assert `ret.value` is not 0

Reply:
Hi @etienne-lms 

Thanks for the review,
you meant like this?

```c
ret = sbi_ecall(SBI_EXT_BASE, SBI_EXT_BASE_PROBE_EXT, extid);
if (!ret.error) {
        assert(ret.value != 0);
        return ret.value;
}
```

If `SBI_EXT_DBCN` is not supported, shouldn't it return 0 (false) so that `sbi_console_ops.putc` can be assigned to `sbi_console_putc_legacy()`?

Reply:
> you meant like this?

yes

> If `SBI_EXT_DBCN` is not supported, shouldn't it return 0 (false) so that `sbi_console_ops.putc` can be assigned to `sbi_console_putc_legacy()`?

I'm not sure to understand the question but I think yes, that's what this ecall is expected to return.

Reply:
debug console extension wasn't supported by OpenSBI until v1.3, and there are other SBI implementation may not support such extension as well, in this case, we may want to fall back to `sbi_console_putc_legacy()`, instread of freeze at the assertion here.

Reply:
So, when `ret.error == 0` and `ret.value == 0`, it means the SBI ID is not supported, right. If so then fine.

Reply:
> So, when `ret.error == 0` and `ret.value == 0`, it means the SBI ID is not supported

Yes, that's correct behavior. Thanks!